compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -520,9 +520,14 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    node->setRegister(source);
 
 
-   //printf("Node %p is first child\n", firstChild); fflush(stdout);
-   //traceMsg(comp, "Node %p is first child\n", firstChild);
-   cg->decReferenceCount(firstChild);
+   if (shouldGenerateGuardedLoadsForCompressedLoad)
```

Comment:
Is this test sufficient?  Looking at the code above that reaches past the `firstChild` to evaluate it's firstChild, its gated with a `firstChild->containsCompressionSequence()` check.  Why is that not needed here?  There seems to be a case when `TR::Compiler->om.compressedReferenceShift() == 0` that will take you through this path as well.  Perhaps I don't understand all the subtle interactions behind these trees for concurrent scavenge, but on the surface there appears to be a small hole in the logic.